### DISCUSSION TITLE
How to use Valtio in a Node.js only environment?

### DISCUSSION THREAD

**Discussion by leonsilicon** [pmndrs/valtio#623] [Q&A]
I've been trying to use Valtio alongside [Ink](https://github.com/vadimdemedes/ink), which is a library for creating Terminal UIs using React. However, it looks like the `useSyncExternalStore` shim Valtio depends on breaks reactivity when used in a server environment (Node.js).

Currently, I have to patch `use-sync-external-store` and hardcode the variable `isServerEnvironment` to false; I'm wondering if there's a better solution? (e.g. the custom [`useZustand` hook](https://npm.im/use-zustand) for zustand)

**Comment by dai-shi** [comment]
```js
import { proxy, snapshot } from 'valtio/vanilla'
```
should work. (but some environments still may complain?)

**Reply by leonsilicon** [reply]
Is there an equivalent for the `useSnapshot` hook? (I'm using valtio in some React/Ink components as well)

**Reply by dai-shi** [reply]
Oh, I misunderstood your question. It's still React.
I think patching use-sync-external-store is a good workaround. (essentially, it's their issue, right?)

I wonder if Ink supports React 18, it doesn't require the shim?

**Reply by leonsilicon** [reply]
Yeah, they don't support React 18 yet (https://github.com/vadimdemedes/ink/issues/526), and I suspect it's not going to be supported for a while (the repo has been mostly dead for a couple of months now), so I guess I'll stick with a patch. Thanks for replying so quickly!
